sql/moz-fx-data-shared-prod/marketing_suppression_list_external/campaign_monitor_suppression_list_v1/query.py

```diff
@@ -0,0 +1,137 @@
+"""Suppression List downloaded from CampaignMonitor API and uploaded to BigQuery."""
+
+import logging
+from pathlib import Path
+from typing import Any, Dict, List
+
+import requests
+import rich_click as click
+import yaml
+from google.cloud import bigquery
+from requests.auth import HTTPBasicAuth
+
+DEFAULT_PROJECT_ID = Path(__file__).parent.parent.parent.name
+DEFAULT_DATASET_ID = Path(__file__).parent.parent.name
+DEFAULT_TABLE_NAME = Path(__file__).parent.name
+
+BASE_URL = "https://api.createsend.com/api/v3.3/clients"
+SCHEMA_FILE = Path(__file__).parent / "schema.yaml"
+SCHEMA = bigquery.SchemaField.from_api_repr(
+    {"name": "root", "type": "RECORD", **yaml.safe_load(SCHEMA_FILE.read_text())}
+).fields
+
+
+def get_api_response(url: str, api_key: str):
+    """Get json of response if the API request was successful."""
+    headers = {"Accept": "application/json"}
+    auth = HTTPBasicAuth(api_key, "")
+
+    response = requests.get(url=url, headers=headers, auth=auth)
+    response.raise_for_status()
+    return response.json()
+
+
+def store_data_in_bigquery(data, schema, destination_project, destination_table_id):
+    """Upload data to Bigquery in a single, non partitioned table."""
+    client = bigquery.Client(project=destination_project)
+
+    job_config = bigquery.LoadJobConfig(
+        create_disposition="CREATE_IF_NEEDED",
+        schema=schema,
+        source_format=bigquery.SourceFormat.NEWLINE_DELIMITED_JSON,
+        write_disposition="WRITE_TRUNCATE",
+    )
+
+    load_job = client.load_table_from_json(
+        data, destination_table_id, location="US", job_config=job_config
+    )
+    load_job.result()
+    stored_table = client.get_table(destination_table_id)
+    logging.info(f"Loaded {stored_table.num_rows} rows into {destination_table_id}.")
+
+
+def download_suppression_list(
+    base_url: str,
+    client_id: str,
+    api_key: str,
+    page_size: int = 1000,
+    order_field: str = "date",
+    order_direction: str = "desc",
```

Comment:
When sorting them by date descending, if a new entry is added mid-sync it could change which page some results end up on, possibly resulting in duplicate records being fetched.  I'd suggest sorting by date ascending.
```suggestion
    order_direction: str = "asc",
```

Reply:
uh good catch!